format/FlightSql.proto

```diff
@@ -867,6 +867,151 @@ enum SqlSupportsConvert {
   SQL_CONVERT_VARCHAR = 19;
 }
 
+enum DATE_SUBCODE {
+  UNKNOWN_SUBCODE = 0;
+  SQL_CODE_DATE = 1;
+  SQL_CODE_TIME = 2;
+  SQL_CODE_TIMESTAMP = 3;
+}
+
+enum SQL_DATA_TYPE {
+  UNKNOWN = 0;
```

Comment:
Similarly, maybe `SQL_CODE_UNKNOWN`, `SQL_CODE_YEAR`, etc.

Reply:
For what it's worth, I'm referencing the Protobuf style guide: https://developers.google.com/protocol-buffers/docs/style#enums 

We haven't been following it *too* strictly but I think we can try to be a little more consistent.